webdriver/tests/interop/beforeunload_prompt.py

```diff
@@ -41,15 +39,16 @@ async def check_beforeunload_not_implicitly_accepted(accept):
         await subscribe_events([USER_PROMPT_CLOSED_EVENT, USER_PROMPT_OPENED_EVENT])
 
         # Using WebDriver classic's navigation command to navigate away from
-        # the page will hang and wait for the beforeunload dialog to close.
-        # As such start the command immediately as task but await for it later
-        # when BiDi closed the prompt.
+        # the page can hang if the prompt is not accepted. As such, wrap the
+        # command to fail on timeout gracefully.
         def sync_navigate():
             current_session.url = page_target
 
-        task_navigate = asyncio.create_task(execute_as_async(sync_navigate))
-        opened_event = await wait_for_future_safe(on_prompt_opened)
+        await wait_for_future_safe(
+            asyncio.create_task(execute_as_async(sync_navigate)))
```

Comment:
The async code shouldn't be needed anymore given that with the `beforeunload` prompt automatically be accepted the WebDriver classic navigation will continue and finally return when the navigation is done.

Reply:
If the user agent does not implement the `beforeunload` accept logic, the test will hang and the whole test will timeout without a meaningful message.

Reply:
Fine with me then.